plugins/gcov/gcov.rake

```diff
@@ -444,35 +193,42 @@ namespace UTILS_SYM do
 
     # Remove unsupported reporting utilities.
     if !(opts[:gcov_utilities].nil?)
-      opts[:gcov_utilities].reject! { |item| !([UTILITY_NAME_GCOVR].map(&:upcase).include? item.upcase) }
+      opts[:gcov_utilities].reject! { |item| !(UTILITY_NAMES.map(&:upcase).include? item.upcase) }
     end
 
     # Default to gcovr when no reporting utilities are specified.
     if opts[:gcov_utilities].nil? || opts[:gcov_utilities].empty?
       opts[:gcov_utilities] = [UTILITY_NAME_GCOVR]
     end
 
-    # Default to HTML basic report when no report types are defined.
-    if opts[:gcov_reports].nil? && opts[:gcov_html_report_type].nil? && opts[:gcov_xml_report].nil?
-      opts[:gcov_reports] = [REPORT_TYPE_HTML_BASIC]
-
-      puts "In your project.yml, define one or more of the"
-      puts "following to specify which reports to generate."
-      puts "For now, creating only an #{REPORT_TYPE_HTML_BASIC} report."
-      puts ""
-      puts ":gcov:"
-      puts "  :reports:"
-      puts "    - #{REPORT_TYPE_HTML_BASIC}"
-      puts "    - #{REPORT_TYPE_HTML_DETAILED}"
-      puts "    - #{REPORT_TYPE_TEXT}"
-      puts "    - #{REPORT_TYPE_COBERTURA}"
-      puts "    - #{REPORT_TYPE_SONARQUBE}"
-      puts "    - #{REPORT_TYPE_JSON}"
-      puts ""
+    if opts[:gcov_reports].nil?
+      opts[:gcov_reports] = []
     end
 
+    gcovr_reportinator = GcovrReportinator.new(@ceedling)
+    gcovr_reportinator.support_deprecated_options(opts)
+
     if is_utility_enabled(opts, UTILITY_NAME_GCOVR)
-      make_gcovr_reports(opts)
+      shell_result = nil
+      total_time = Benchmark.realtime do
+        shell_result = gcovr_reportinator.make_reports(opts)
+      end
+      if shell_result
+        shell_result[:time] = total_time
+        print_shell_result(shell_result)
```

Comment:
@aschlicht Thanks for review! As you can see it is much easier to notice something in someone other one code than in my own.
I've added reportinator helper with method to report results.
I would be appreciated if you check my changes, because you have spend much more time with these changes.

BTW I've noticed that you removed first empty line in ReportGeneratorReportinator class. I'm not sure. I just noticed this coding style in other ceedling files.

Reply:
@CezaryGapinski I did remove that first blank line. If other files keep the first line blank, then let's be consistent and restore the blank line. I do like consistency.

Reply:
@CezaryGapinski I reviewed the latest changes with reportinator helper and I think they are good. Runs as I'd expect too. Thanks for your continued help!